doc/main.dox

```diff
@@ -172,13 +172,27 @@ is to use a camera-specific variant of a subtask. Examples include:
 - lsst.obs.sdss.selectSdssImages.SelectSdssImagesTask: an version of the task that selects images
     for co-addition of SDSS stripe 82 images
 
-Here is an example of retargeting a subtask in a config override file; this retargets
-lsst.pipe.tasks.exampleCmdLineTask.ExampleCmdLineTask `stats` with a simpler version:
+How you retarget a subtask depends if the subtask is specified as an
+\ref lsst.pex.config.configurableField.ConfigurableField "lsst.pex.config.ConfigurableField"
+(the most common case, and the source of the term "retarget") or as an
+\ref lsst.pex.config.registry.RegistryField "lsst.pex.config.RegistryField".
+Here is an example of the former; this retargets lsst.pipe.tasks.exampleCmdLineTask.ExampleCmdLineTask
+`stats` with a simpler version:
 <pre>
     from lsst.pipe.tasks.exampleStatsTasks import ExampleSimpleStatsTask
     config.stats.retarget(ExampleSimpleStatsTask)
 </pre>
 
+If a subtask is specified as an
+\ref lsst.pex.config.registry.RegistryField "lsst.pex.config.RegistryField"
+then the subtask is retargeted by setting the `name` property of the field
+to the name of the desired subtask, for example:
+
+<pre>
+    import ...  # the package in which subtask "bar" is defined, so it can register itself
+    config.aRegistrySubtask.name = "bar"
+</pre>
+
```

Comment:
If by "as above" you mean the same as used in the "how to write a task" documentation then I agree and will update it accordingly. I will also include an example of overriding configuration parameters.